server/lib/availabilitySync.ts

```diff
@@ -57,141 +54,93 @@ class AvailabilitySync {
 
         // We can not delete media so if both versions do not exist, we will change both columns to unknown or null
         if (!mediaExists) {
+          logger.info(
+            `Media with TMDB ID ${media.tmdbId} does not exist in any of your media instances. Status will be changed to unknown.`,
+            { label: 'AvailabilitySync' }
+          );
+
+          // Find all related requests only if
+          // related media is available
+          const requests = await requestRepository
+            .createQueryBuilder('request')
+            .leftJoinAndSelect('request.media', 'media')
+            .where('(media.id = :id)', {
+              id: media.id,
+            })
+            .andWhere(
+              '((request.is4k = 0 AND media.status IN (:...mediaStatus)) OR (request.is4k = 1 AND media.status4k IN (:...mediaStatus)))',
+              {
+                mediaStatus: [
+                  MediaStatus.AVAILABLE,
+                  MediaStatus.PARTIALLY_AVAILABLE,
+                ],
+              }
+            )
+            .getMany();
+
+          // Check if a season is processing or pending to
+          // make sure we set media to the correct status
+          let mediaStatus = MediaStatus.UNKNOWN;
+          let mediaStatus4k = MediaStatus.UNKNOWN;
+
+          if (media.mediaType === 'tv') {
+            mediaStatus = await this.findMediaStatus(requests, false);
+            mediaStatus4k = await this.findMediaStatus(requests, true);
+          }
```

Comment:
Switched to using query builder to make sure requests are related to only available media. We then check those requests to set the media to the correct status. For example, if the media is deleted and a request still remains, the media should be set to processing.